### DISCUSSION TITLE
version conflict, document already exists (current version [1])

### DISCUSSION THREAD

**Discussion by Phillip-Samson** [Security-Onion-Solutions/securityonion#11207] [2.4]
### Version

2.4.100

### Installation Method

Network installation on Red Hat derivative like Oracle, Rocky, Alma, etc.

### Description

other (please provide detail below)

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### Network Traffic Collection

other (please provide detail below)

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

Yes, there are additional clues in /opt/so/log/ (please provide detail below)

### Detail

I am getting the following error in my /opt/so/logs/logstash/log file should. Should I be concerned?

[
    2023-08-30T11: 10: 14,
    966
][WARN
][logstash.outputs.elasticsearch
] Failed action {
    :status=>409,
    :action=>[
        "create",
        {
            :_id=>nil,
            :_index=>"logs-sentinel_one.alert-default",
            :routing=>nil
        },
        {
            "message"=>"{\"agentDetectionInfo\":{\"accountId\":\"redacted\",\"machineType\":\"desktop\",\"name\":\"redacted\",\"osFamily\":\"windows\",\"osName\":\"Windows 10 Enterprise\",\"osRevision\":\"19045\",\"siteId\":\"redacted\",\"uuid\":\"redacted\",\"version\":\"22.3.2.373\"},\"agentRealtimeInfo\":{\"id\":\"redacted\",\"infected\":false,\"isActive\":true,\"isDecommissioned\":false,\"machineType\":\"desktop\",\"name\":\"redacted\",\"os\":\"windows\",\"uuid\":\"redacted\"},\"alertInfo\":{\"alertId\":\"1762967095372986067\",\"analystVerdict\":\"Undefined\",\"createdAt\":\"2023-08-30T10:20:33.754000Z\",\"dnsRequest\":null,\"dnsResponse\":null,\"dstIp\":null,\"dstPort\":null,\"dvEventId\":\"01H92ZAXMVB29FF8CRTGSP6YVS_52\",\"eventType\":\"HTTP\",\"hitType\":\"Events\",\"incidentStatus\":\"Unresolved\",\"indicatorCategory\":null,\"indicatorDescription\":null,\"indicatorName\":null,\"isEdr\":true,\"loginAccountDomain\":null,\"loginAccountSid\":null,\"loginIsAdministratorEquivalent\":null,\"loginIsSuccessful\":null,\"loginType\":null,\"loginsUserName\":null,\"modulePath\":null,\"moduleSha1\":null,\"netEventDirection\":null,\"registryKeyPath\":null,\"registryOldValue\":null,\"registryOldValueType\":null,\"registryPath\":null,\"registryValue\":null,\"reportedAt\":\"2023-08-30T10:20:54.939204Z\",\"source\":\"STAR\",\"srcIp\":null,\"srcMachineIp\":null,\"srcPort\":null,\"tiIndicatorComparisonMethod\":null,\"tiIndicatorSource\":null,\"tiIndicatorType\":null,\"tiIndicatorValue\":null,\"updatedAt\":\"2023-08-30T10:20:54.939204Z\"},\"containerInfo\":{\"id\":null,\"image\":null,\"labels\":null,\"name\":null},\"kubernetesInfo\":{\"cluster\":null,\"controllerKind\":null,\"controllerLabels\":null,\"controllerName\":null,\"namespace\":null,\"namespaceLabels\":null,\"node\":null,\"pod\":null,\"podLabels\":null},\"ruleInfo\":{\"description\":\"Block URLs associated with Windows Defender popups\",\"id\":\"1705951576678980835\",\"name\":\"Windows Defender Browser Hijack 3\",\"queryLang\":\"1.0\",\"queryType\":\"events\",\"s1ql\":\"Url Contains Anycase \\\"ondigitalocean.app/?number=\\\" OR Url Contains Anycase \\\"ondigitalocean.app/werrx01\\\" OR URL Contains Anycase \\\"diholke38b.tk\\\" OR Url Contains Anycase \\\"z13.web.core.windows.net\\\" OR Url Contains Anycase \\\"bigassfans.com\\\" OR Url Contains Anycase \\\"159.65.38.240\\\" OR Url Contains Anycase \\\"asdcaw.za.com/?gclid=\\\" OR Url Contains Anycase \\\"pendulumdisd.store/?gclid\\\" OR Url Contains Anycase \\\"nbdfv.za.com/?gclid=\\\" OR Url Contains Anycase \\\"gum.criteo.com/syncframe?origin=publishertag\\u0026topUrl=www.oceandraw.com\\\" OR Url Contains Anycase \\\"foodonlinebest.xyz/Win0HelplCHine00Smt01\\\" OR Url Contains Anycase \\\"alert-ca-nw-xx-smc-c-cdcdc-cklmll-dczdcexs-s-sse-ws-ekeeszz.azurewebsites.net\\\" OR Url Contains \\\"/win0xxm08/?phone=\\u0026\\\" OR Url Contains Anycase \\\"cedefoilg.com\\\" OR Url Contains Anycase \\\"s3.eu-west-3\\\" OR Url Contains Anycase \\\"herofherlittl.com\\\" OR Url Contains Anycase \\\"oughsheukwa.autos\\\" OR Url Contains Anycase \\\"ewbwebe3827g3-secondary.z13.web.core.windows.net\\\" OR Url Contains Anycase \\\"ittfk56s-f444svmk7a-uc.a.run.app\\\"\",\"scopeLevel\":\"account\",\"severity\":\"Low\",\"treatAsThreat\":\"Malicious\"},\"sourceParentProcessInfo\":{\"commandline\":\"\\\"C:\\\\Program Files (x86)\\\\Microsoft\\\\Edge\\\\Application\\\\msedge.exe\\\" --no-startup-window --win-session-start /prefetch:5\",\"fileHashMd5\":\"63aa0d78b269dda6ce41f79ced2d6d97\",\"fileHashSha1\":\"54288b9b5d170fb90aa9769a28050b35e5b3fd7a\",\"fileHashSha256\":\"50a1f7dcf6c1a1a78c13914ca9e4c3cc6650a3d512560142e108345127c9372b\",\"filePath\":\"C:\\\\Program Files (x86)\\\\Microsoft\\\\Edge\\\\Application\\\\msedge.exe\",\"fileSignerIdentity\":\"MICROSOFT CORPORATION\",\"integrityLevel\":\"medium\",\"name\":\"msedge.exe\",\"pid\":\"12712\",\"pidStarttime\":\"2023-08-30T08:03:33.558000Z\",\"storyline\":\"0830EE540EBD4B9E\",\"subsystem\":\"sys_win32\",\"uniqueId\":\"0730EE540EBD4B9E\",\"user\":\"redacted\\\\redacted\"},\"sourceProcessInfo\":{\"commandline\":\"\\\"C:\\\\Program Files (x86)\\\\Microsoft\\\\Edge\\\\Application\\\\msedge.exe\\\" --no-startup-window /prefetch:5\",\"fileHashMd5\":\"63aa0d78b269dda6ce41f79ced2d6d97\",\"fileHashSha1\":\"54288b9b5d170fb90aa9769a28050b35e5b3fd7a\",\"fileHashSha256\":\"50a1f7dcf6c1a1a78c13914ca9e4c3cc6650a3d512560142e108345127c9372b\",\"filePath\":\"C:\\\\Program Files (x86)\\\\Microsoft\\\\Edge\\\\Application\\\\msedge.exe\",\"fileSignerIdentity\":\"MICROSOFT CORPORATION\",\"integrityLevel\":\"medium\",\"name\":\"msedge.exe\",\"pid\":\"13996\",\"pidStarttime\":\"2023-08-30T10:04:24.660000Z\",\"storyline\":\"0830EE540EBD4B9E\",\"subsystem\":\"sys_win32\",\"uniqueId\":\"redacted\",\"user\":\"redacted\\\\redacted\"},\"targetProcessInfo\":{\"tgtFileCreatedAt\":\"1970-01-01T00:00:00Z\",\"tgtFileHashSha1\":null,\"tgtFileHashSha256\":null,\"tgtFileId\":null,\"tgtFileIsSigned\":\"signed\",\"tgtFileModifiedAt\":\"1970-01-01T00:00:00Z\",\"tgtFileOldPath\":null,\"tgtFilePath\":null,\"tgtProcCmdLine\":null,\"tgtProcImagePath\":null,\"tgtProcIntegrityLevel\":\"unknown\",\"tgtProcName\":null,\"tgtProcPid\":null,\"tgtProcSignedStatus\":null,\"tgtProcStorylineId\":null,\"tgtProcUid\":null,\"tgtProcessStartTime\":\"1970-01-01T00:00:00Z\"}}",
            "@version"=>"1",
            "input"=>{
                "type"=>"httpjson"
            },
            "metadata"=>{
                "version"=>"8.8.2",
                "input_id"=>"httpjson-sentinel_one-redacted",
                "input"=>{
                    "beats"=>{
                        "host"=>{
                            "ip"=>"redacted"
                        }
                    }
                },
                "raw_index"=>"logs-sentinel_one.alert-default",
                "stream_id"=>"httpjson-sentinel_one.alert-redacted",
                "beat"=>"filebeat",
                "type"=>"_doc"
            },
            "event"=>{
                "created"=>"2023-08-30T11:10:03.432Z",
                "dataset"=>"sentinel_one.alert"
            },
            "agent"=>{
                "name"=>"orsonfnodep01",
                "id"=>"redacted",
                "version"=>"8.8.2",
                "ephemeral_id"=>"redacted",
                "type"=>"filebeat"
            },
            "@timestamp"=>2023-08-30T11: 10: 03.432Z,
            "type"=>"redis-input",
            "elastic_agent"=>{
                "id"=>"redacted",
                "version"=>"8.8.2",
                "snapshot"=>false
            },
            "tags"=>[
                "forwarded",
                "sentinel_one-alert",
                "elastic-agent",
                "input-orseconip01",
                "beats_input_codec_plain_applied"
            ],
            "ecs"=>{
                "version"=>"8.0.0"
            },
            "cloud"=>{
                "region"=>"",
                "provider"=>"huawei",
                "availability_zone"=>"iad-ad-1",
                "service"=>{
                    "name"=>"ECS"
                },
                "instance"=>{
                    "id"=>"ocid1.instance.oc1.iad.anuwcljrbcao5qicrl432flzzh6udbk7z5jcmtmqvlksysjdf6v75pabynta"
                }
            },
            "data_stream"=>{
                "namespace"=>"default",
                "type"=>"logs",
                "dataset"=>"sentinel_one.alert"
            }
        }
    ],
    :response=>{
        "create"=>{
            "_index"=>".ds-logs-sentinel_one.alert-default-2023.08.24-000003",
            "_id"=>"9PDR9spJ3coNnHrpEJFoFadt7IA=",
            "status"=>409,
            "error"=>{
                "type"=>"version_conflict_engine_exception",
                "reason"=>"[9PDR9spJ3coNnHrpEJFoFadt7IA=]: version conflict, document already exists (current version [1])",
                "index_uuid"=>"er_hYdQtQGWNTXLuRrY0aQ",
                "shard"=>"0",
                "index"=>".ds-logs-sentinel_one.alert-default-2023.08.24-000003"
            }
        }
    }
}

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by weslambert** [comment]
Do you only see this message for SentinelOne data?

**Reply by Phillip-Samson** [reply]
The message is in the SentinelOne Index with that specific ID

**Reply by Phillip-Samson** [reply]
Any thoughts on this topic, I am now seeing this issue with Darktrace SentinelOne and Okta. 
version conflict, document already exists
